CMakeLists.txt

```diff
@@ -87,6 +87,9 @@ if(${USE_Z3_DLOPEN})
 elseif (${Z3_FOUND})
   add_definitions(-DHAVE_Z3)
   message("Z3 SMT solver found. This enables optional SMT checking with Z3.")
+  if ("${Z3_VERSION}" VERSION_LESS "4.8.12.0")
+    message(WARNING "SMT tests need Z3 4.8.12 or higher, SMT tests are expected to fail!")
+  endif()
```

Comment:
Not sure if we want this warning in a general case. In the PR I mentioned a warning that would be issued only when your build includes `isoltest` and only if you have older Z3 installed but disabled with `-DUSE_Z3=OFF`.

@leonardoalt What do you think? Are we fine with a broad warning like this one?

Reply:
I can make it more precise to print the warning if TESTS are ON.

Reply:
@cameel Sorry I missed your message. Yea I think you're right that we shouldn't output this in the general case. I'll take a look at the current code.

Reply:
@leonardoalt What do you think about what @soroosh-sdi proposed in https://github.com/ethereum/solidity/pull/11873#issuecomment-913024021?

I think that's actually pretty neat. I thought we'll have to make it depend on build targets but that actually solves the problem without too much magic.

@soroosh-sdi One thing I'd change is that the warning should only be printed if the library is not found. `TESTS` is enabled by default so otherwise we'll always be seeing it.

Reply:
Yea I think that sounds good too. We only have to update that a couple times a year, so the benefit is bigger than the overhead I'd say.